src/main/java/org/veupathdb/service/eda/ds/util/RFileSetProcessor.java

```diff
@@ -18,40 +19,56 @@
     public final String name;
     public final InputStream stream;
     public final Optional<Integer> maxAllowedRows;
+    public final Boolean showMissingness;
+    public final List<String> nonStrataColNames;
     public final ConsumerWithException<RConnection> fileReader;
 
-    public RFileProcessingSpec(String name, InputStream stream, Optional<TwoTuple<Optional<Integer>,BiConsumerWithException<String, RConnection>>> processingInfo) {
+    public RFileProcessingSpec(String name, InputStream stream, Boolean showMissingness, List<String> nonStrataColNames, Optional<TwoTuple<Optional<Integer>,BiConsumerWithException<String, RConnection>>> processingInfo) {
       this.name = name;
       this.stream = stream;
       this.maxAllowedRows = processingInfo.map(i -> i.getFirst()).orElse(Optional.empty());
+      this.showMissingness = showMissingness;
+      this.nonStrataColNames = nonStrataColNames;
       this.fileReader = conn -> processingInfo.map(i -> i.getSecond()).orElse((a,b) -> {}).accept(name, conn);
     }
   }
 
   private final Map<String,InputStream> _dataStreams;
```

Comment:
I would put all this stuff in an object and use a single map; before when there were only two things I used a TwoTuple, but with four we should just make a little container object and put it in a single map (so there would be two total maps: one for InputStreams, one for config)

Reply:
im not opposed. i even considered it when i was working on this fri. i guess i didnt understand why it was a TwoTuple to begin. (im sure theres a reason, i just dont know it.) why couldnt/shouldnt these each just be their own map, with four (showMissingness, nonStrataColNames, maxAllowedRows, fileReader) total currently?

Reply:
also, my son came home sick from school today and my time is limited. given we already have something implemented, do you think its worth merging without this? or do you know if the front end guys already have enough to work with if i sit on this for a bit longer?